tests/dataset_test.py

```diff
@@ -123,6 +123,81 @@ def test_del_item_missing():
         del d['not an item']
 
 
+def test_update_from_dict_adds_items():
+    a = sc.DataArray(sc.scalar(1.0))
+    b = sc.DataArray(sc.scalar(2.0))
+    b2 = sc.DataArray(sc.scalar(3.0))
+    c = sc.DataArray(sc.scalar(4.0))
+    d = sc.Dataset({'a': a, 'b': b})
+    d.update({'b': b2, 'c': c})
+    assert sc.identical(d['a'], a)
+    assert sc.identical(d['b'], b2)
+    assert sc.identical(d['c'], c)
+
+
+def test_update_from_dataset_adds_items():
+    a = sc.DataArray(sc.scalar(1.0))
+    b = sc.DataArray(sc.scalar(2.0))
+    b2 = sc.DataArray(sc.scalar(3.0))
+    c = sc.DataArray(sc.scalar(4.0))
+    d = sc.Dataset({'a': a, 'b': b})
+    d.update(sc.Dataset({'b': b2, 'c': c}))
```

Comment:
Could parametrize the test on the arg of `update`?

Reply:
Do you mean in order to combine all the different tests? Not sure if this would be more readable.

Reply:
At least some of them? In some cases the difference is tiny.